### PR TITLE
[onert] Add Package of NNFW Python API

### PR DESCRIPTION
**Pull request by KangInyeong**
This commit adds code for making and publishing nnfwapi package including NNFW Python API module.

for #11368 

ONE-DCO-1.0-Signed-off-by: KangInyeong kiyoog02@gmail.com
Co-authored-by: jaemaning lovesay00@naver.com
Co-authored-by: phj799 phj799@naver.com
Co-authored-by: seohhh shy1624@naver.com
Co-authored-by: blue4683 an4683@gmail.com
Co-authored-by: jeonghwanin jhpc0128@naver.com

### CODE REVIEW

**Review by hseok-oh — APPROVED** [review]

File: infra/nnfw/python/setup.py
Comment (ragmani) [inline infra/nnfw/python/setup.py:17]:
```suggestion
    # remove architecture directory
```

File: infra/nnfw/python/setup.py
Comment (ragmani) [inline infra/nnfw/python/setup.py:31]: @hseok-oh 
Is there a good way to avoid using relative path?

Comment (hseok-oh) [reply infra/nnfw/python/setup.py:31]: I don't good at python packaging. I just checked python code itself.

**Review by ragmani — DISMISSED** [review]
LGTM (Thanks for your contribution)

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

**Comment by KangInyeong** [comment]
@nnfw-bot test onert-x64-release

File: infra/nnfw/python/nnfwapi/libnnfw_api_pybind.py
Comment (glistening) [inline infra/nnfw/python/nnfwapi/libnnfw_api_pybind.py:26]: Error message is too general. Anyway, later, someone may improve the error message.

File: infra/nnfw/python/pack.sh
Comment (glistening) [inline infra/nnfw/python/pack.sh:3]:
```suggestion
# package
```

File: infra/nnfw/python/pack.sh
Comment (glistening) [inline infra/nnfw/python/pack.sh:7]: I guess `pack.sh` is not for typical user, but for maintainer. Also, `pack.sh` requires `twine` installed.

File: infra/nnfw/python/setup.py
Comment (glistening) [inline infra/nnfw/python/setup.py:15]: Why do we need `for` loop? Why don't you remove `packaging_directory` at once?

File: infra/nnfw/python/setup.py
Comment (glistening) [reply infra/nnfw/python/setup.py:31]: It would be good to remove relative path. I guess this code may not work if `setup.py` is executed out of the same directory.

If so, what about using current script directory as starting point? I searched and 
 `os.path.dirname(os.path.abspath(__file__))` returns the directory.

File: infra/nnfw/python/pack.sh
Comment (KangInyeong) [reply infra/nnfw/python/pack.sh:7]: Yes, `pack.sh` is written for maintainer. I will add `sudo apt install twine` on README.md for packaging.

File: infra/nnfw/python/setup.py
Comment (glistening) [inline infra/nnfw/python/setup.py:55]:
```suggestion
    long_description='It provides nnfw Python api',
```

**Review by glistening — APPROVED** [review]
You may update the comment in another PR.

File: infra/nnfw/python/nnfwapi/libnnfw_api_pybind.py
Comment (KangInyeong) [reply infra/nnfw/python/nnfwapi/libnnfw_api_pybind.py:26]: First, I'll change error message to include more details!

**Comment by KangInyeong** [comment]
@nnfw-bot test onert-x64-release

File: infra/nnfw/python/setup.py
Comment (KangInyeong) [reply infra/nnfw/python/setup.py:15]: Except for the directories in "packaging_directory", some files like setup.py and pack.sh were intended not to be deleted..! 
Do you think it's better to write code in a way other than `for` loop?

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

**Review by hseok-oh — APPROVED** [review]

**Review by ragmani — APPROVED** [review]

File: infra/nnfw/python/setup.py
Comment (glistening) [reply infra/nnfw/python/setup.py:15]: I don't understand the aim of this PR, maybe due to my lack of knowledge about python packaging. I expected this PR is for script to generate whl, but it seems not.

File: infra/nnfw/python/README.md
Comment (glistening) [inline infra/nnfw/python/README.md:14]: It does not work, maybe
```
$ pip install -i https://test.pypi.org/simple/ nnfwapi
```

I got this command from #11595.

Comment (ragmani) [reply infra/nnfw/python/README.md:14]: They tested using testpypi, not pypi.

Comment (ragmani) [reply infra/nnfw/python/README.md:14]: They tested using testpypi. We will have to upload it later to pypi due to account issues.
I think it is correct to write the document as above. Do you think it would be better to update this document later as well?

File: infra/nnfw/python/setup.py
Comment (KangInyeong) [reply infra/nnfw/python/setup.py:15]: I didn't fully understand your comment saying that  "README.md requires to install prebuilt python package".
By using `setup.py` with `python3 setup.py sdist bdist_wheel` command in `pack.sh`, nnfwapi package is created. It makes the so files into a deployable state. After uploading this package using`twine`, anyone can install and use packages through PyPI without having to build themselves.

File: infra/nnfw/python/README.md
Comment (KangInyeong) [reply infra/nnfw/python/README.md:14]: As @ragmani said, we used TestPyPI for the test of the packaging process. In fact, we thought it should be uploaded to PyPI based on this case in the future. `README.md` was written in the form of 'pip install <package's name>' assuming that it will be uploaded to pypi. Perhaps, `pack.sh` written for convenient work may have been confused because the TestPyPI code is active instead of PyPI for uploading trhough `twine` .

Comment (glistening) [reply infra/nnfw/python/README.md:14]: I was aware of your plan. But it does not work. I am not sure when it will be uploaded in official pypi. I think it needs to be updated and revisited before uploading official site. It is the reason I asked the working version. I didn't know how to install if #11595 did not exist.

File: infra/nnfw/python/setup.py
Comment (glistening) [reply infra/nnfw/python/setup.py:15]: > ## Requirement
>
> `nnfwapi` should be installed from PyPI.
> ```
> $ pip install nnfwapi
> ```

I just followed the guide of `README.md`. It requires to install `nnfwapi`.

File: infra/nnfw/python/pack.sh
Comment (glistening) [inline infra/nnfw/python/pack.sh:4]: `pack.sh` does 2 different tasks: `packaging` and `publishing`. I think they should be in different script.

Someone may want to make a package, but don't want to publish.

File: infra/nnfw/python/nnfwapi/__init__.py
Comment (glistening) [inline infra/nnfw/python/nnfwapi/__init__.py:2]: Why `infra` needs to have this python script?

File: infra/nnfw/python/pack.sh
Comment (KangInyeong) [reply infra/nnfw/python/pack.sh:4]: `pack.sh` is written for doing the package and publishing easily.
I think there are two options:
1) Change `pack.sh`  In the form of receiving options from the user 
     ex. for packaging, use `sh pack.sh -p` / for uploading, use `sh pack.sh -u`
2) Delete `pack.sh` shell script and provide information for each command at `README.md`

Comment (glistening) [reply infra/nnfw/python/pack.sh:4]: For this PR, both work for me. Please choose what you feel convinient. I will get opinion frome others and update if necessary.

File: infra/nnfw/python/nnfwapi/__init__.py
Comment (jeonghwanin) [reply infra/nnfw/python/nnfwapi/__init__.py:2]: We have libnnfw_api_pybind.py because

It help user import libnnfw_api_pybind that matches the architecture automatically.
It serves as the interface between libnnfw_api_pybind.so and the user. (nnfw_session, tensorinfo)
We used this approach of placing different SO files with the same name in architecture-specific folders to distinguish them.
Is there an alternative approach available for achieving this?

Comment (glistening) [reply infra/nnfw/python/nnfwapi/__init__.py:2]: I am okay for the content of `*.py` themselves. I wonder why it is located under `infra`.

Comment (glistening) [reply infra/nnfw/python/nnfwapi/__init__.py:2]: > **init**.py, libnnfw_api_pybind.py need to be part of a package. How about moving **init**.py, libnnfwa_api_pybind.py in 'ONE/runtime/onert/python/'?

I don't take a look in detail. But if it is the code for typical user, it would be better be a part of python package, and put somewhere under `runtime/onert/python`, not under `infra`.

Comment (jeonghwanin) [reply infra/nnfw/python/nnfwapi/__init__.py:2]: Ok, thanks for your guidance,
We will update

**Review by glistening — APPROVED** [review]
You may update my comment in another PR. It already has 2 approvals and my comment is not critical.

**Comment by ragmani** [comment]
@KangInyeong Do you want to wait until @jeonghwanin approves this PR? If not, please remove @jeonghwanin from the reviewer request list.

**Comment by KangInyeong** [comment]
> @KangInyeong Do you want to wait until @jeonghwanin approves this PR? If not, please remove @jeonghwanin from the reviewer request list.

No such a wait is not necessary. Let's check the reviewer settings.
And then I'll make additional PR for the updated code based on comments on PR...!

**Review by jeonghwanin — APPROVED** [review]

**Comment by ragmani** [comment]
Sorry for the incorrect information. I didn't know it cannot be removed.

**Pull request merged**
